local-notifications/ios/Plugin/LocalNotificationsDelegate.swift

```diff
@@ -0,0 +1,87 @@
+import Capacitor
+import UserNotifications
+
+public class LocalNotificationsDelegate: NSObject, NotificationHandlerProtocol {
+
+    public var plugin: CAPPlugin?
+
+    // Local list of notification id -> JSObject for storing options
+    // between notification requets
+    var notificationRequestLookup = [String: JSObject]()
+
+    public func requestPermissions(with completion: ((Bool, Error?) -> Void)? = nil) {
+        let center = UNUserNotificationCenter.current()
+        center.requestAuthorization(options: [.badge, .alert, .sound]) { (granted, error) in
+            completion?(granted, error)
+        }
+    }
+
+    public func checkPermissions(with completion: ((Bool, Error?) -> Void)? = nil) {
+        let center = UNUserNotificationCenter.current()
+        center.requestAuthorization(options: [.badge, .alert, .sound]) { (granted, error) in
```

Comment:
this will request the permissions, I don't think that's expected on checkPermissions

Reply:
I missed my own TODO 😅